cekit/docgen.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import logging
+import os
+import yaml
+import sys
+from jinja2 import Environment, Template
+
+logger = logging.getLogger('cekit')
+
+autogen_warning="""////
+    AUTOGENERATED FILE - this file was generated via "cekit docgen".
+    Changes to .adoc or HTML files may be overwritten! To make changes,
+    please change cekit's template.
+////
+
+"""
+
+class Docgen():
+    def __init__(self):
+        # find module descriptor
+        self.descriptor = 'module.yaml'
+        if not os.path.isfile(self.descriptor):
+            logger.error("can't find module.yaml descriptor")
+            sys.exit(1)
+        logger.debug("module descriptor path {} ".format(self.descriptor))
+
+        # set up template path
+        self.template_file = os.path.join(os.path.dirname(__file__),
+                                          'templates',
+                                          'template.adoc.jinja')
+        logger.debug("module doc template file {} ".format(self.template_file))
+
+    def docgen(self):
+        self.template = Template(open(self.template_file).read())
+        self.generate_doc_for_module(self.descriptor)
+
+    module_dirs = ['./jboss']
+
+    def generate_doc_for_module(self, module_file):
+        with open(module_file) as open_file:
+            data = yaml.load(open_file)
+        output_file = os.path.join(os.path.dirname(module_file), 'README.adoc')
+        print ("Generating %s..." % os.path.join(os.path.relpath(output_file, os.getcwd())), os.path.basename(output_file))
+        with open(output_file, "w") as text_file:
+            text_file.write(autogen_warning)
+            text_file.write(self.template.render(data))
+
+    def scan_for_modules(self, curdir=os.getcwd()):
+        if not os.path.isdir(curdir):
+            return
```

Comment:
I would use sys.exit(0)

Reply:
I agree, although I think I might eventually rewrite this whole routine around `os.walk`

Reply:
I might cut this entirely for this PR and possibly bring it back in a later one. It's still not hooked into the "docgen" verb anyway.